### ISSUE TITLE
Laser cut svg

### ISSUE THREAD

**Issue by dattanchu** [tbfleming/jscut#15]
I couldn't find the required settings to configure jscut for laser cutting along the path instead of outside the path.

**Comment by tbfleming** [comment]
The Engrave operation should work. Let me know.

**Comment by tbfleming** [comment]
I accidentally broke that option on an earlier commit.

**Comment by dattanchu** [comment]
Wouldn't engrave be different from laser cut along the path? Is there something I can help looking into this matter? If so, can I get some pointers?

**Comment by tbfleming** [comment]
engrave follows the path as-is; the other options modify or fill the path. Right now jscut doesn't know how to turn a laser on and off, but you can add the appropriate commands to the generated gcode.

**Comment by tbfleming** [comment]
Summary from https://groups.google.com/forum/#!topic/jscut/W3dNRqbDIcM:
- Abdenour Saaid will try laser cutting with the engraving option
- He will have to manually insert M3 and M5 into the gcode
- If it works I may add an option to automatically insert M3 and M5

I invite others to try this and report back either here or in the jscut Google Group.

**Comment by tbfleming** [comment]
No one has reported back, which indicates lack of interest. I'll close this for now.

**Comment by ghost** [comment]
Seconding interest... from several people in fact (two hackerspaces, one signshop and my own laser)

Will fire up the test code (where you attempt to control the laser using the spindle pins), just waiting for my tiny-tiny-g boards to arrive
In the meantime - the LasaurGrbl would need (and I want to port most of this over to TinyG as the way it works is quite simplistically elegant):

Sxx command to set intensity:   As you select multiple objects, you'd want to select different power levels for each - so use case, I have a square with some words on it.  1.  Select the words, set up a engraving operation @10% power to just burn the text onto the piece, at say 3000 feed.  2  Select the square and do a "outside" cut (compensate for the beam kerf by using a 0.1mm 'tool') at 100% power to ensure you cut through, and set a feedrate of 1500

In terms of Lasersaur firmware based boards, G1's are cuts and G0 = moves - much simpler than adding M3 and M5 (see why I want to port it into tinyG? or even just keep the M328p')

So an example job:

G90 ;   Absolute Coordinates
M80 ;   Turn on Air Valve 
G0F8000 ; Set feedrate for non-cut moves
G1F1500 ;  Set feedrate for cut-moves
S77 ;   Set laser intensity (77/225 = 30%) for the first item to cut
G0X113.70Y74.61  ; .... Just move without cutting to the start of the first object
G1X113.45Y78.05  ;  Start the cut (G1 moves from here...
G1X112.90Y80.71
G1X112.06Y83.28
G1X110.95Y85.73

**Comment by tbfleming** [comment]
Reopening for discussion.

I was unaware of LasaurApp before yesterday; it provides a web interface for both generating the paths and running the machine. It's also specialized for lasers. Why duplicate their effort?

It looks like this will complicate jscut several ways:
- Laser mode needs a different gcode generator; the above example breaks multiple assumptions inside jscut's current gcode generator.
- Laser mode needs a different simulator gcode parser
- Every operation needs both an intensity and a speed setting for laser cutting. These will be ignored for milling.
- Pocket will look ugly; laser really needs raster scanning.
- These mill settings are worthless for laser cutting:
  - Operation: Ramp Plunge
  - Operation: Direction
  - Operation: Cut Depth
  - Operation: Width
  - Tabs: Max Cut Depth
  - Tool: Angle
  - Tool: Pass Depth
  - Tool: Step Over
  - Tool: Plunge
  - Mat: Thickness
  - Mat: Z Origin
  - Mat: Clearance

**Comment by peteruithoven** [comment]
@tbfleming I agree with you, it doesn't make sense to combine the tools. Why not try to make everything as modular as possible and share building blocks? (I understand this is lot's of work)

**Comment by mirage335** [comment]
LasaurApp seems most appropriate for that specifc laser cutter. Some of us with RepRap laser cutters would like to use something like jscut instead.

**Comment by ghost** [comment]
While Lasaurapp works, its not perfect for my workflow (since I use Marlin (;  - see http://www.openbuilds.com/builds/openbuilds-freeburn-2-mini-v-slot-co2-laser-40-50w.1657/

Moreover though,   @tbfleming I hear you using the stupid open source mess of DXF files as a continued excuse but Lasaurapp does manage to import any DXF I create (Qcad, Sketchup Plugin) and if it doesnt, a file-> save as in QCad is all it takes...

**Comment by tbfleming** [comment]
Lasuarapp doesn't have to offset polygons to do cutter compensation, so it doesn't need to care about closed loops. jscut does offset polygons, so it completely falls apart without closed loops. The offsetting is so fundamental to all of jscut's algorithms (cutter comp is just the easiest to explain) that it's never disabled.

**Comment by mirage335** [comment]
And, offsetting is actually relevant to precision laser cutting, even required, even with high-precision UV lasers.

**Comment by cheton** [comment]
Just to mention that there is a gcode post processor for jscut that can convert gcode for simple laser cutting.
https://github.com/eraclitux/to-laser

**Comment by ghost** [comment]
Thanks Cheton. Useful!
